addons/product/models/product.py

```diff
@@ -610,9 +613,34 @@ def _has_valid_attributes(self, valid_attributes, valid_values):
             :return: True if the attibutes and values are correct, False instead
         """
         self.ensure_one()
-        values = self.attribute_value_ids.filtered(lambda v: v.attribute_id.create_variant != 'no_variant')
+        values = self.attribute_value_ids
         attributes = values.mapped('attribute_id')
-        return attributes == valid_attributes and values <= valid_values
+        if attributes != valid_attributes:
```

Comment:
return attributes == valid_attributes and any([v in valid_calues for v v in values])

Reply:
Do you think it has a significant impact on performance or readability to write it this way?

Because it makes test coverage more difficult to read: with one line, the line is always covered no matter the result.

But when writing multiple lines with conditions, it's possible to make sure every single case is actually covered. This has been really useful to me when writing tests or making sure the condition worked as it should.